plugins/techdocs/src/reader/components/Reader.tsx

```diff
@@ -66,20 +65,30 @@ const useEnforcedTrailingSlash = (): void => {
   }, []);
 };
 
-export const Reader = () => {
+type Props = {
+  componentId: {
+    kind: string;
+    namespace: string;
+    name: string;
+  };
+};
+
+export const Reader = ({ componentId }: Props) => {
   useEnforcedTrailingSlash();
 
+  const { kind, namespace, name } = componentId;
+  const { '*': path } = useParams();
+
   const docStorageUrl =
     useApi(configApiRef).getOptionalString('techdocs.storageUrl') ??
     'https://techdocs-mock-sites.storage.googleapis.com';
 
-  const location = useLocation();
-  const { componentId, '*': path } = useParams();
   const [shadowDomRef, shadowRoot] = useShadowDom();
   const navigate = useNavigate();
   const normalizedUrl = new URLFormatter(
-    `${docStorageUrl}${location.pathname.replace('/docs', '')}`,
+    `${docStorageUrl}/${kind}/${namespace}/${name}`,
```

Comment:
😀 Would it be possible to make this overrideable via an apiRef interface? That way we can still use this with our pre-existing docs storage & build setup.

Reply:
Yea that's definitely something we should add. Just having storageBase/name could cause issues though since the name is not a unique identifier. I'd recommend you either store by ID or this combination of kind, namespace and name.

Reply:
Thanks! In our case the name is the repo name, so it will be unique. We have pre-exiting CI steps to publish to our storage location based on the repo name. Having the ability to override this will avoid us having to use the existing convention with minimal changes.